core/shared/src/main/scala/scalaz/zio/ZIO.scala

```diff
@@ -1586,6 +1596,42 @@ trait ZIOFunctions extends Serializable {
    */
   final def descriptor: UIO[Fiber.Descriptor] = ZIO.Descriptor
 
+  /**
+   * Checks the interrupt status, and produces the effect returned by the
+   * specified callback.
+   */
+  final def checkInterruptible[R >: LowerR, E <: UpperE, A](f: Boolean => ZIO[R, E, A]): ZIO[R, E, A] =
+    new ZIO.CheckInterrupt(f)
```

Comment:
Not `descriptor >>= (d => d.interruptible)`?

Reply:
I thought about that (indeed, that's why I added `interruptible` to `Descriptor`), but it's becoming more and more expensive to call `descriptor` (it allocates a lot, accesses volatile memory, etc.). All this logic will appear in the new `ensuring`, which was previously a primitive, so I'm trying to keep an eye on performance issues now in advance of benchmarking/optimization.